infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -99,71 +101,103 @@ public LogReplicationMetadataManager(CorfuRuntime rt, long topologyConfigId, Str
     }
 
     public CorfuStoreMetadata.Timestamp getTimestamp() {
-        return corfuStore.getTimestamp();
+        Token token = runtime.getSequencerView().query().getToken();
+        return CorfuStoreMetadata.Timestamp.newBuilder()
+                .setEpoch(token.getEpoch())
+                .setSequence(token.getSequence())
+                .build();
```

Comment:
The only caller is `processOpaqueEntry`, so we can remove this function completely.